lib/msf/core/post_mixin.rb

```diff
@@ -49,6 +49,12 @@ def setup
       # for its platform, capabilities, etc.
       check_for_session_readiness if session.type == "meterpreter"
 
+      if session.type.ends_with?(':winpty')
+        raise Msf::OptionValidateError.new({
+          'SESSION' => 'Session does not support post modules.'
+        })
+      end
+
```

Comment:
Heh :wink: 
Maybe a #TODO for someone digging around to address this down the line?